weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksCallRefExpansion.ts

```diff
@@ -71,98 +82,350 @@ export const useClientSideCallRefExpansion = (
+          }
+
+          // If we found the path, recursively replace refs in it
+          if (target !== undefined && parent !== null) {
+            // Replace refs in the target value
+            replaceAllRefsRecursively(target, refsDataMap);
+          }
+        });
+      } else {
+        // Original implementation: only replace refs in specified columns
+        const expandedRefColumnsList = Array.from(expandedRefColumns ?? []);
+        expandedRefColumnsList.forEach(col => {
+          const colParts = col.split('.');
+          let value: any = call;
+          const path: string[] = [];
+          for (const part of colParts) {
+            while (
+              typeof value === 'object' &&
+              value != null &&
+              EXPANDED_REF_VAL_KEY in value
+            ) {
+              value = value[EXPANDED_REF_VAL_KEY];
+              path.push(EXPANDED_REF_VAL_KEY);
+            }
+            if (value == null) {
+              break;
+            }
+            if (typeof value !== 'object' || !(part in value)) {
+              value = null;
+              break;
+            }
+            value = value[part];
+            path.push(part);
+          }
+          while (
+            typeof value === 'object' &&
+            value != null &&
+            EXPANDED_REF_VAL_KEY in value
+          ) {
+            value = value[EXPANDED_REF_VAL_KEY];
+            path.push(EXPANDED_REF_VAL_KEY);
+          }
+          if (isWeaveRef(value) && refsDataMap.has(value)) {
+            const refObj = refsDataMap.get(value);
+            _.set(call, path, makeRefExpandedPayload(value, refObj));
+          }
+        });
       }
-      while (
-        typeof value === 'object' &&
-        value != null &&
-        EXPANDED_REF_VAL_KEY in value
+
+      return call;
+    });
+
+    return doExpansionIteration(
+      expandedTraceCalls,
+      expandedRefColumns,
+      client,
+      recursiveUnwrap,
+      iterationCount + 1,
+      maxIterations
+    );
+  } catch (error) {
+    // Return the calls as they are if an error occurs
+    return traceCalls;
+  }
+};
+
+/**
+ * Recursively finds all expandable references in an object
+ * @param obj The object to search
+ * @param refsNeeded Set to collect found references
+ * @param visited Set of objects already visited to prevent circular reference issues
+ * @param depth Current recursion depth
+ * @param maxDepth Maximum recursion depth to prevent stack overflow
+ */
+const findAllRefsRecursively = (
+  obj: any,
+  refsNeeded: Set<string>,
+  visited: Set<any> = new Set(),
+  depth: number = 0,
+  maxDepth: number = 50
+): void => {
+  // Prevent excessive recursion
+  if (depth > maxDepth) {
+    return;
+  }
+
+  // Base case: null or primitive value
+  if (!obj || typeof obj !== 'object') {
+    if (isExpandableRef(obj)) {
+      refsNeeded.add(obj);
+    }
+    return;
+  }
+
+  // Prevent circular references
+  if (visited.has(obj)) {
```

Comment:
i wonder if this blows up for large objects, we could hash it for better mem perf at the cost of cpu time